keps/582-preempt-based-on-flavor-order/README.md

```diff
@@ -165,52 +165,73 @@ proposal will be implemented, this is the place to discuss them.
 
 ### Cluster Queue API
 
-We extend the Cluster Queue API to introduce the new fields: flavorFungibility to opt-in and configure the new behavior.
+We extend the Cluster Queue API to introduce the new field `flavorFungibility` to opt-in and configure the behavior, and a minimal preference field for choosing borrowing-first vs preemption-first when the scheduler is exploring multiple flavors.
 
-For each type of resource in each podSet, Kueue will traverse all resource groups and resource flavors to find a available flavor in present. When there are insufficient resources in the flavor, kueue will prioritize preemption or borrowing based on the configured policy. 
+For each type of resource in each podSet, Kueue will traverse all resource groups and resource flavors to find an available flavor in present. When there are insufficient resources in the flavor, Kueue will prioritize preemption or borrowing based on the configured policy and, when applicable, the preference.
 
 ```
 const (
-	Borrow        FlavorFungibilityPolicy = "Borrow"
-	Preempt       FlavorFungibilityPolicy = "Preempt"
-	MayStopSearch FlavorFungibilityPolicy = "MayStopSearch"
-	TryNextFlavor FlavorFungibilityPolicy = "TryNextFlavor"
+    Borrow        FlavorFungibilityPolicy = "Borrow"        // deprecated alias of MayStopSearch
+    Preempt       FlavorFungibilityPolicy = "Preempt"       // deprecated alias of MayStopSearch
+    MayStopSearch FlavorFungibilityPolicy = "MayStopSearch"
+    TryNextFlavor FlavorFungibilityPolicy = "TryNextFlavor"
+)
+
+type FlavorAssignmentPreference string
```

Comment:
```suggestion
type FlavorFungibilityPreference string
```

I would recommend using `FlavorFungibility` or `FlavorFungibilityAssignment` prefix to avoid conflict in the future.